aixplain/processes/data_onboarding/labelstudio_onboard.py

```diff
@@ -0,0 +1,331 @@
+            else:
+                return {'Error': 'No annotations'}
+        elif 'url' in dict['data']:
+            output['data'] = dict['data']['url']
+            output['id'] = dict['id']
+            output['annotation'] = []
+            prev_start = -1
+            prev_end = -1
+            dtypes = {}
+            try:
+                if 'annotations' in dict and 'result' in dict['annotations'][0]:
+                    item = {}
+                    for annotation in dict['annotations'][0]['result']:
+                        name = annotation['from_name']
+                        annotation_type = annotation['type']
+                        if annotation['type'] == 'textarea':
+                            annotation_type = 'text'
+                        tags = annotation['value'][annotation_type][0]
+                        dtypes[name] = annotation['type']
+                        if 'start' in annotation['value']:
+                            start = annotation['value']['start']
+                        if 'end' in annotation['value']:
+                            end = annotation['value']['end']
+                        if prev_start != start and len(item) > 0:
+                            item['start'] = prev_start
+                            item['end'] = prev_end
+                            output['annotation'].append(item)
+                            item = {}
+                        if 'start' in annotation['value']:
+                            item[name] = tags
+                            prev_start = start
+                            prev_end = end
+                    if len(item) > 0:
+                        item['start'] = prev_start
+                        item['end'] = prev_end
+                        output['annotation'].append(item)
+            except:
+                print('Error: No annotations found!')
+            return output, dtypes
+    else:
+        return {'Error': 'No data'}
+
+def getAllDataForAProject(url, header, project_id):
+    """
+    Retrieves and extracts data for all tasks associated with a specific project in a Label Studio instance.
+
+    Args:
+        url (str): The base URL of Label Studio.
+        header (dict): The request headers containing authentication details.
+        project_id (int): The ID of the project for which tasks' data are to be retrieved.
+
+    Returns:
+        tuple: A tuple containing two dictionaries. The first dictionary contains extracted data and annotations,
+               while the second dictionary maps annotation names to their corresponding data types.
+    """
+    output = []
+    tasks = getAllTasksPerProject(url, header, project_id)
+    dtypes = extractData(json.loads(getTaskData(url, header, tasks[0])))[1]
+    for task in tasks:
+        output.append(extractData(json.loads(getTaskData(url, header, task)))[0])
+    return output, dtypes
+
+def auto_onboard(
+    labelstudio_key: str,
+    data_name: str,
```

Comment:
Instead of "data", use `corpus` to refer to the data collection so that:

```
data_name -> corpus_name
data_description -> corpus_description
```